Sources/BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -528,8 +539,18 @@ extension CryptoStore: BraveWalletKeyringServiceObserver {
     rejectAllPendingWebpageRequests()
   }
   public func keyringCreated(_ keyringId: String) {
+    Task { @MainActor [weak self] in
+      if let newCoin = WalletConstants.supportedCoinTypes.first(where: { $0.keyringId == keyringId }) {
+        self?.userAssetManager.migrateUserAssets(for: newCoin)
+      }
+    }
   }
   public func keyringRestored(_ keyringId: String) {
+    // if a keyring is restored, we want to reset user assets local storage
+    // and migrate with for new keyring
+    WalletUserAssetGroup.removeAllGroup() { [weak self] in
+      self?.userAssetManager.migrateUserAssets()
```

Comment:
Same for restoring a wallet (fresh install):

https://github.com/brave/brave-ios/assets/5314553/676a9bed-315f-474e-a465-0a98a5952288